doc/faq.rst

```diff
@@ -458,3 +458,87 @@ How do I set a ``random_state`` for an entire execution?
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Please refer to :ref:`randomness`.
+
+Why do categorical variables need preprocessing in scikit-learn, compared to other tools?
+-----------------------------------------------------------------------------------------
+
+Most of scikit-learn assumes data is in NumPy arrays or SciPy sparse matrices
+of a single numeric dtype. These do not explicitly represent categorical
+variables at present. Thus, unlike R's data.frames or pandas.DataFrame, we
+require explicit conversion of categorical features to numeric values, as
+discussed in :ref:`preprocessing_categorical_features`.
+See also :ref:`sphx_glr_auto_examples_compose_plot_column_transformer_mixed_types.py` for an
+example of working with heterogeneous (e.g. categorical and numeric) data.
+
+Why does Scikit-learn not directly work with, for example, pandas.DataFrame?
+----------------------------------------------------------------------------
+
+The homogeneous NumPy and SciPy data objects currently expected are most
+efficient to process for most operations. Extensive work would also be needed
+to support Pandas categorical types. Restricting input to homogeneous
+types therefore reduces maintenance cost and encourages usage of efficient
+data structures. But :class:`~sklearn.compose.ColumnTransformer`
+can be used with pandas DataFrame because this meta estimator makes it convenient
+to map related subsets of dataframe columns selected by name or dtype to 
+dedicated scikit-learn transformers.
```

Comment:
Actually all scikit-learn estimators can be used with pandas dataframe (as long as they have homogeneous column types). What `ColumnTransformer` offers is a way to deal
with dataframes with non-homogeneous columns. Here is an alternative phrasing to avoid inducing any confusion in that respect:

```suggestion
data structures. 

Note however that :class:`~sklearn.compose.ColumnTransformer` makes it
convenient to handle heterogeneous pandas dataframes by mapping homogeneous subsets of
dataframe columns selected by name or dtype to dedicated scikit-learn transformers.

Therefore :class:`~sklearn.compose.ColumnTransformer` are often used in the first
step of scikit-learn pipelines when dealing
with heterogeneous dataframes (see :ref:`pipeline` for more details).

See also :ref:`sphx_glr_auto_examples_compose_plot_column_transformer_mixed_types.py`
for an example of working with heterogeneous (e.g. categorical and numeric) data.
```